synapse/models/auth.py

```diff
@@ -1,14 +1,60 @@
+import hashlib
+
+import synapse.lib.types as s_types
+
+class Passwd(s_types.Str):
+
+    def norm(self, valu):
+        retn = s_types.Str.norm(self, valu)
+        retn[1].setdefault('subs', {})
+        byts = retn[0].encode('utf8')
+        retn[1]['subs'].update({
+            'md5': hashlib.md5(byts, usedforsecurity=False).hexdigest(),
+            'sha1': hashlib.sha1(byts, usedforsecurity=False).hexdigest(),
+            'sha256': hashlib.sha256(byts).hexdigest(),
+        })
+        return retn
+
 modeldefs = (
+
     ('auth', {
+
+        'ctors': (
+            ('auth:passwd', 'synapse.models.auth.Passwd', {'strip': False}, {
+                'interfaces': (
+                    ('auth:credential', {}),
+                    ('crypto:hashable', {}),
+                ),
+                'doc': 'A password string.'}),
+        ),
+
         'types': (
+
             ('auth:credential', ('ndef', {'interfaces': ('auth:credential',)}), {
                 'doc': 'An ndef type including all forms which implement the auth:credential interface.'}),
         ),
+
         'interfaces': (
             ('auth:credential', {
                 'doc': 'An interface inherited by authentication credential forms.',
             }),
         ),
-        'forms': (),
+
+        'forms': (
+            ('auth:passwd', {}, (
+                ('md5', ('crypto:hash:md5', {}), {
```

Comment:
Worth making these virts on the type?

Reply:
I think pivoting will be fine for this use case :+1: